pycape/cape.py

```diff
@@ -149,14 +158,14 @@ def encrypt(
             Exception: if the enclave threw an error while trying to fulfill the
                 connection request.
         """
-        cape_key = key or self.key(key_path)
+        cape_key = key or await self.key(key_path)
         ctxt = cape_encrypt.encrypt(input, cape_key)
         # cape-encrypted ctxt must be b64-encoded and tagged
         ctxt = base64.b64encode(ctxt)
         return b"cape:" + ctxt
 
-    @contextlib.contextmanager
-    def function_context(self, function_ref: Union[str, fref.FunctionRef]):
+    @_synchronizer.asynccontextmanager
+    async def function_context(self, function_ref: Union[str, fref.FunctionRef]):
```

Comment:
hmmm have you experimented with `@_synchronizer.asynccontextmanager`? When I run `python invoke_echo.py` example, I am getting:
```
Traceback (most recent call last):
  File "invoke_echo.py", line 31, in <module>
    with cape.function_context(function_ref):
AttributeError: __enter__
```

Reply:
fixed :+1: ty